package exception.trycatch;

public class TryCatchExample1 {

	public static void main(String[] args) {
		
		int i = 10;
		int j = 0;
		
		System.out.println(i * j);
		try {
		System.out.println(i / j); //0으로 나누면 error가 발생
		//0만 아니면 error가 아니다. 이런 코드를 에러발생가능성이있는 코드
		//빨간줄이 안뜨면 error인지 예측하기 어렵다. 돌려봐야 안다. 
		System.out.println("에외 발생시 이 텍스트는 출력되지 않음.");
		}catch(Exception e) {
			System.out.println("0으로 나눌 수 없습니다.");
			//error발생하는 순간 다음 코드는 실행하지 않고 바로 catch로 넘어간다.
		}
		System.out.println(i + j);
		System.out.println(i - j);

	}
	/*
	 * 개발자의 경험에 의해서 예외 처리 코드를 삽입해야 합니다. 
	 * 주요 실행 예 4가지 
	 * 
	 * 
	 */

}
